compiler/rustc_span/src/span_encoding.rs

```diff
@@ -89,14 +89,15 @@ impl Span {
 
         let (base, len, ctxt2) = (lo.0, hi.0 - lo.0, ctxt.as_u32());
 
-        if len <= MAX_LEN && ctxt2 <= MAX_CTXT && parent.is_none() {
+        if len <= MAX_LEN && ctxt2 < MAX_CTXT && parent.is_none() {
             // Inline format.
-            Span { base_or_index: base, len_or_tag: len as u16, ctxt_or_zero: ctxt2 as u16 }
+            Span { base_or_index: base, len_or_tag: len as u16, ctxt_or_max: ctxt2 as u16 }
         } else {
             // Interned format.
             let index =
                 with_span_interner(|interner| interner.intern(&SpanData { lo, hi, ctxt, parent }));
-            Span { base_or_index: index, len_or_tag: LEN_TAG, ctxt_or_zero: 0 }
+            let ctxt_or_max = if ctxt2 < MAX_CTXT { ctxt2 } else { MAX_CTXT } as u16;
```

Comment:
`MAX_CTXT` is now a misleading name -- it's actually one past the max context, because it's a special out-of-band value. Perhaps rename it `INTERNED_CTXT`? And likewise rename `ctxt_or_max` as `ctxt_or_interned`?